src/nvim/api/ui.c

```diff
@@ -340,7 +345,69 @@ void nvim_ui_pum_set_height(uint64_t channel_id, Integer height, Error *err)
                   "It must support the ext_popupmenu option");
     return;
   }
+
+  ui->pum_nlines = (int)height;
+}
+
+/// Tells Nvim the geometry of the popumenu, to align floating 
+/// windows with an external popup menu. Note that this method
+/// is not to be confused with |nvim_ui_pum_set_height()|, which
+/// sets the number of visible items in the popup menu, while
+/// this function sets the bounding box of the popup menu, 
+/// including visual decorations such as boarders and sliders.
+///
+/// @param channel_id
+/// @param width   Popupmenu width, must be greater than zero.
+/// @param height  Popupmenu height, must be greater than zero.
+/// @param row     Popupmenu row, must be greater or equal to zero.
+/// @param col     Popupmenu height, must be greater or equal to zero.
+/// @param[out] err Error details, if any. On error, suspend pum position reporting for the current UI.
+void nvim_ui_pum_set_bounds(uint64_t channel_id, Integer width, Integer height, 
```

Comment:
Float positioning can use floating point values (floats need not use the same font size, nor be anchored to exact grid corers). So probably this should use `Float` as well (otherwise it will be a gap if the pum occupies 10.5 cells in width, etc).

Reply:
Does that mean `pum_set_event_info` should also pack floats into the dict?

Reply:
Testing the "float" build to see if it works like before.

Reply:
Backend plugins are happy with it. Frontend needs to send floats over msgpack, otherwise get type errors. Adjusted `fvim` and it works.

Reply:
Nope, have to test again with `tv_dict_add_float`

Reply:
After changing event argument to float, backend plugin `coc.nvim` interprets geometry as (0,0,0,0)

Reply:
All good.

Reply:
Hmm, ideally we should accept a msgpack int as a Float argument (we have similar rules for `bool` and handles already). it would be annoying if `ui_pum_set_bounds(1, ...)` in a dynamically typed client errors out.

Reply:
done.